### PR TITLE
Feature/dp 221 organisation lookup both name and identifier

### PR DESCRIPTION
**Pull request by JBaigGoaco**
image

### CODE REVIEW

File: Services/CO.CDP.Organisation.WebApi.Tests/Extensions/ServiceCollectionExtensionsTests.cs [hunk 1]
```diff
@@ -78,7 +78,7 @@ public void MapException_Should_Return_BadRequest_For_InvalidQueryException()
         var result = ServiceCollectionExtensions.MapException(exception);
 
         Assert.Equal(StatusCodes.Status400BadRequest, result.status);
-        Assert.Equal("MISSING_QUERY_PARAMETERS", result.error);
+        Assert.Equal("ISSUE_WITH_QUERY_PARAMETERS", result.error);
```
Comment (jakzal) [inline Services/CO.CDP.Organisation.WebApi.Tests/Extensions/ServiceCollectionExtensionsTests.cs:81]: Why changing this test instead of adding a new one? 🤔

Comment (jakzal) [reply Services/CO.CDP.Organisation.WebApi.Tests/Extensions/ServiceCollectionExtensionsTests.cs:81]: I see, you changed the code for the existing case.

File: Services/CO.CDP.Organisation.WebApi/UseCase/LookupOrganisationUseCase.cs [hunk 2]
```diff
@@ -9,6 +9,11 @@ public class LookupOrganisationUseCase(IOrganisationRepository organisationRepos
 {
     public async Task<Model.Organisation?> Execute(OrganisationQuery query)
     {
+        if (!string.IsNullOrEmpty(query.Name) && !string.IsNullOrEmpty(query.Identifier))
+        {
+            throw new InvalidQueryException("Both name and identifier cannot be provided together.");
+        }
```
Comment (jakzal) [inline Services/CO.CDP.Organisation.WebApi/UseCase/LookupOrganisationUseCase.cs:15]: Can we add a test case for this please.

Comment (JBaigGoaco) [reply Services/CO.CDP.Organisation.WebApi/UseCase/LookupOrganisationUseCase.cs:15]: testcase is added

**Review by jakzal — APPROVED** [review]

**Pull request merged**